monkey/monkey_island/cc/ui/src/components/types/MapNode.tsx

```diff
@@ -0,0 +1,34 @@
+export class MapNode {
+    machine_id: number
+    operating_system: string
+    hostname: string
+    network_interfaces: object
+    agent_is_running: boolean
+    island: boolean
+    propagated_to: boolean
+    connections: object
+    agent_id?: string
+    parent_id?: string
+
+    constructor(machine_id: number,
+        operating_system: string,
+        hostname: string,
+        network_interfaces: object,
+        agent_is_running: boolean,
+        island: boolean,
+        propagated_to: boolean,
+        connections: object,
+        agent_id?: string,
+        parent_id?: string) {
```

Comment:
What's the point of these being optional?

Reply:
If we scan a machine that hasn't been exploited yet, it won't have an agent or parent ID